### PR TITLE
Fix compose offset when keyboard pops up.

### PR DESCRIPTION
**Pull request by vishwesh3**
Fix:#212
screenshot_1490379074

### CODE REVIEW

File: src/chat/Chat.js [hunk 1]
```diff
@@ -17,26 +24,30 @@ export default class MainScreen extends React.Component {
...
+            auth={auth}
+            fetchOlder={fetchOlder}
+            fetchNewer={fetchNewer}
+            doNarrow={doNarrow}
+            markAsRead={markAsRead}
+            mute={mute}
+          />
+        </View>
+        <View style={moreStyles.composeBox}>
+          {canSendToNarrow(narrow) && <ComposeBox onSend={this.sendMessage} />}
```
Comment (borisyankov) [inline src/chat/Chat.js]: Are the extra Views necessary?
The one around MessageList may not be.
And the one around ComposeBox - you probably can apply the styles directly to the ComposeBox component?

Comment (vishwesh3) [inline src/chat/Chat.js]: @borisyankov Done

**Comment by vishwesh3** [comment]
**Update** (https://github.com/zulip/zulip-mobile/pull/457/commits/3ac77e8d70d81a1d73a7f4650524c0d90dbdd524)

Now input expands as soon as text becomes multiline.
ezgif com-video-to-gif 13

**Comment by borisyankov** [comment]
Just tested the latest version. There is a small issue:
The navigation bar moves off-screen when the user focuses on the compose box.

**Comment by vishwesh3** [comment]
I have done changes in the `android/app/src/main/AndroidManifest.xml`.
So to affect those changes, we need to re-install app. (`react-native run-android`)

**Comment by borisyankov** [comment]
I tested this on iOS and it doesn't work, which is expected because the xml change is Android specific.

I would have thought a purely JS solution is possible, and the KeyboardAvoidingView should be part of the solution.

**Comment by borisyankov** [comment]
@vishwesh3 Let's focus on this task now. See if you can find a universal solution, and I promise to review quickly. We would need that fixed for first app store release.

**Comment by vishwesh3** [comment]
We want to keep navigation bar as it is when keyboard pop's up, so `adjustPan` cannot be used, as we don't want to adjust the whole pan. I found `adjustResize` most appropriate as it resizes the activity (our RN app is under this; `MainActivity`) window to make room for the soft keyboard. Keyboard pop's up is handled by android activity (which is the parent and our RN app (code) is a child) and it makes room, now there is no need of KeyboardAvoidingView (for android). If we still use it then 
ezgif com-video-to-gif 31
it will again give padding.

**PR updated**
for now, we can go with `View` in Android and `KeyboardAvoidingView` in iOS, and do a release. And we can keep experimenting to get a simple solution.

Now it looks like this
on iOS
ezgif com-video-to-gif 33


on Android

ezgif com-video-to-gif 32

**Comment by vishwesh3** [comment]
@zulipbot thanks for the reminder. Updated :+1:

**Comment by vishwesh3** [comment]
@zulipbot thanks for the reminder! Updated PR 👍

**Comment by borisyankov** [comment]
Hey @vishwesh3 let's get this PR merged next.
Please rebase.
I am not sure the iOS version works so we definitely want this to work both on iOS and Android, and we strongly should prefer having the same code for both.

**Comment by kunall17** [comment]
I just tested this PR so one problem still persist while the soft keyboard opens up we want the chat messages to shift upside and make space for the keyboard which is common in every other app!

**Comment by vishwesh3** [comment]
>the soft keyboard opens up we want the chat messages to shift upside and make space for the keyboard 

I think we only want to shift messages up, if list is scrolled to end and keyboard pop's up. 
This also exists on iOS. We can work on this next.
ezgif com-video-to-gif-5

**Comment by vishwesh3** [comment]
Updated 👍 
Just found that our issue is somewhat similar to
https://github.com/facebook/react-native/issues/11681

How it looks now?
ezgif com-video-to-gif-6

**Comment by borisyankov** [comment]
Thanks, @vishwesh3 
That was so much work to fix that 🥇

**Pull request merged**